core/src/main/java/org/mskcc/cbio/oncokb/util/GenomeNexusUtils.java

```diff
@@ -215,18 +227,43 @@ private static String getIsoform(Gene gene, ReferenceGenome referenceGenome) {
         }
     }
 
-    private static VariantAnnotation getVariantAnnotation(GNVariantAnnotationType type, String query, ReferenceGenome referenceGenome) throws ApiException {
-        VariantAnnotation variantAnnotation = null;
-        if (query != null && type != null) {
+    public static String chromosomeNormalizer(String chromosome) {
+        return chromosome.trim().replace("chr", "").replace("23", "X").replace("24", "Y");
+    }
+
+    private static GenomicLocation convertGenomicLocation(String genomicLocation) {
+        if (genomicLocation == null) {
+            return null;
+        }
+        String[] parts = genomicLocation.split(",");
+        GenomicLocation location = null;
+        if (parts.length >= 5) {
+            // trim all parts
+            for (int i = 0; i < parts.length; i++) {
+                parts[i] = parts[i].trim();
+            }
+            location = new GenomicLocation();
+            location.setChromosome(chromosomeNormalizer(parts[0]));
+            location.setStart(parts[1].isEmpty() ? null : Integer.parseInt(parts[1]));
+            location.setEnd(parts[2].isEmpty() ? null : Integer.parseInt(parts[2]));
+            location.setReferenceAllele(parts[3]);
+            location.setVariantAllele(parts[4]);
+        }
+        return location;
+    }
+
+    private static List<VariantAnnotation> getVariantsAnnotation(GNVariantAnnotationType type, List<String> queries, ReferenceGenome referenceGenome) throws ApiException {
+        List<VariantAnnotation> variantsAnnotation = null;
+        if (queries != null && type != null) {
             List<String> gnFields = new ArrayList<>();
             gnFields.add("annotation_summary");
             if (type.equals(GNVariantAnnotationType.HGVS_G)) {
-                variantAnnotation = getAnnotationControllerApi(referenceGenome).fetchVariantAnnotationGET(query, MSK_ISOFORM_OVERRIDE, null, gnFields);
+                variantsAnnotation = getAnnotationControllerApi(referenceGenome).fetchVariantAnnotationPOST(queries, MSK_ISOFORM_OVERRIDE, null, gnFields);
             } else {
-                variantAnnotation = getAnnotationControllerApi(referenceGenome).fetchVariantAnnotationByGenomicLocationGET(query, MSK_ISOFORM_OVERRIDE, null, gnFields);
+                variantsAnnotation = getAnnotationControllerApi(referenceGenome).fetchVariantAnnotationByGenomicLocationPOST(queries.stream().map(GenomeNexusUtils::convertGenomicLocation).collect(Collectors.toList()), MSK_ISOFORM_OVERRIDE, null, gnFields);
```

Comment:
If `convertGenomicLocation` returns null, is that supported by the GN POST request? I used the GN swagger to make a POST with an array `[{valid GL object}, null]` and I get a 404 error. We should filter out the null entries?

Reply:
My thinking is we should not. We've been trying to persist the query location in the list. WI'm thinking to return null for the null input.

Reply:
Gotcha, persisting the index is valid. However, do you know if passing in a null value in the POST body array will throw an exception for the entire request even if there are valid queries to annotate?

Reply:
null in request body or in response?